osf-cas/templates/configmap.yaml

```diff
@@ -386,6 +380,37 @@ config/cas.properties: |-
   cas.logout.remove-descendant-tickets=false
   ########################################################################################################################
 
+  ########################################################################################################################
+  # OSF URLs
+  ########################################################################################################################
+  # OSF
+  #
+  cas.authn.osf-url.home=https://{{ .Values.osfDomain }}/
+  cas.authn.osf-url.dashboard=https://{{ .Values.osfDomain }}/dashboard/
+  cas.authn.osf-url.login-with-next=https://{{ .Values.osfDomain }}/login?next=
+  cas.authn.osf-url.logout=https://{{ .Values.osfDomain }}/logout/
+  cas.authn.osf-url.resend-confirmation=https://{{ .Values.osfDomain }}/resend/
+  cas.authn.osf-url.forgot-password=https://{{ .Values.osfDomain }}/forgotpassword/
+  cas.authn.osf-url.register=https://{{ .Values.osfDomain }}/register/
+  ########################################################################################################################
+
+  ########################################################################################################################
+  # OSF API Settings for Institution Authentication
+  ########################################################################################################################
+  # Authentication Endpoint
+  #
+  cas.authn.osf-api.instn-authn-endpoint=https://{{ .Values.apiDomain }}/v2/institutions/auth/
+  #
+  # JWT / JWE secrets for signing and encrypting authentication request payload
+  #
+  cas.authn.osf-api.instn-authn-jwt-secret=${OSF_JWT_SECRET}
+  cas.authn.osf-api.instn-authn-jwe-secret=${OSF_JWE_SECRET}
+  #
+  # Path of the XSL file for parsing and transforming XML authentication responses
+  #
+  cas.authn.osf-api.instn-authn-xsl-location=file:/etc/cas/institutions-auth.xsl
+  ########################################################################################################################
```

Comment:
New chained property class `OsfApiProperties` has been created in the backend to support institution SSO.

Both `OSF_JWT_SECRET` and `OSF_JWE_SECRET` already exists in the private settings (i.e. inherited from oldCAS), thus no change needed in the private settings.

I am not 100% sure that `file:/etc/cas/institutions-auth.xsl` this will work but let's give it a try. Otherwise, we have to modify the private settings. As for staging3, the case-insensitive issue for institution SSO exists but it is not an issue for now since staging3 doesn't do any SSO at all.